packages/components/src/components/DiagramFlamegraph.vue

```diff
@@ -0,0 +1,48 @@
+<template>
+  <div class="diagram-flame">
+    <div ref="d3Container"></div>
+  </div>
+</template>
+
+<script>
+// namespace import to better support commonjs
+import * as d3 from 'd3';
+import * as Flamegraph from 'd3-flame-graph';
+import { digestEventArray } from '../lib/flamegraph';
+
+const { flamegraph: createFlamegraph } = Flamegraph;
+
+export default {
+  name: 'v-diagram-flame',
+
+  props: {
+    theme: {
+      type: String,
+      default: 'dark',
+      validator: (value) => ['dark', 'light'].includes(value),
+    },
+    appMap: {
+      type: Object,
+    },
```

Comment:
I usually prefer to see components require the bare minimum properties. Otherwise we restrict the context in which we can add a flamegraph to other locations (i.e., we're saying here that we'd need an entire AppMap). In this case, `rootEvents` or even a pre-digested event array might be more suitable. I'll leave it up to your discretion.

Reply:
fixup: 36e774af53d38121844c9bd642c85138061999ea and a54ec1c2103bc84bcb427e2dd71e177d3a912590

Reply:
But in order for the filters to work, isn't the `appMap` parameter required?

Reply:
We do want to eventually support flame graphs for huge AppMaps, but not in v0.

Reply:
At the moment it is not required, so I think what Dustin said make sense. We can tweak the property later imo.